File: prueba_facturas/Modelos/Factura.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Driver.Encryption;

namespace prueba_facturas.Modelos

{
    public class Factura
    {

        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)] 
        public string id { get; set; }


        public string CodigoFactura { get; set; }
        
        public Cliente Cliente { get; set; }

        public string Ciudad { get; set; }

        public string Nit { get; set; }

        public decimal TotalFactura { get; set; }

        public decimal SubTotal { get; set; }

        public decimal Iva { get; set; }

        public decimal Retencion { get; set; }

        public DateTime FechaCreacion { get; set; }

        public Estado Estado { get; set; }

        public bool Pagada { get; set; }

        public DateTime? FechaPago { get; set; }

    }
}

File: frontFacturas/src/app/interfaces/factura.ts
export interface factura {
   codigoFactura: string;
   nombreCliente: string;
   ciudad: string;
   nit: string;
   totalFactura: number;
   subTotal: number;
   iva: number;
   retencion: number;
   fechaCreacion: Date;
   estado: Estado;
   pagada: boolean;
   fechaPago: Date | null;
}

export enum Estado {
    primerrecordatorio = 0,
    segundorecordatorio = 1,
    Desactivado = 2,
}